### DISCUSSION TITLE
One click stalwart installation

### DISCUSSION THREAD

**Discussion by antonind72** [coollabsio/coolify#4118] [Service Template Requests]
do you think it would be possible to add stalwart as a one-click application? stalwar is a modern and complete mail server. here are the links :

website: https://stalw.art/
github: https://github.com/stalwartlabs/mail-server

**Comment by dr-diffie** [comment]
That would be an awesome improvement for the Community as One-Click-Service.

Runs easy as hell in the docker container and would offer users to also host their whole mail infrastructure + production code in Coolify to run fully self-hosted. Works securely and automatic only by adjusting your DNS Settings to handle mails. Also wrote a little script for myself to just copy and paste their provided output format to adjust my records on Cloudflare. Works like charm.

**Reply by antonind72** [reply]
> That would be an awesome improvement for the Community as One-Click-Service.
> 
> Runs easy as hell in the docker container and would offer users to also host their whole mail infrastructure + production code in Coolify to run fully self-hosted. Works securely and automatic only by adjusting your DNS Settings to handle mails. Also wrote a little script for myself to just copy and paste their provided output format to adjust my records on Cloudflare. Works like charm.




for my part I tried to configure it ( stalwart )
but I can't

**Reply by dr-diffie** [reply]
I got it working and tested it. Cleaning up the docker compose right now and preparing the pull request over the weekend 🙏🏻

**Reply by francescoamici** [reply]
@dr-diffie  any news about this? i'd love to see it as One-Click-Service!

**Comment by dr-diffie** [comment]
@ysfaran

**Comment by bparlan** [comment]
any update on this?

I've been trying to self-host my mailserver, yet unfortunate so far. Couldn't make any available solutions work (docker-mailserver, mailcow) so far. Somehow when I add Coolify as docker image link, the end situation doesn't let me open admin panel that I get via `docker logs stalwart-mail` . Not sure if I am messing with cloudflare, coolify container names, or somewhere in ssh'd configs...

**Reply by mikolajzawadzki** [reply]
Why do you even need a mail server? Almost all data centers block the ports for sending emails. And even if you manage to find an IP you can send from, most of your emails will end up in spam anyway

**Reply by bparlan** [reply]
All I need is to receive emails tbh, my gmail logs in and checks from all inboxes regardless. issue is I dont wanna pay a hosting yearly money just to receive emails into my domain mailbox while paying VPS here. do you think it is really troublesome just to receive emails? I am kinda at solving the certificates issue part of the story, so feeling like quite close... do you advice any other solution regarding domain address mail?

**Reply by mikolajzawadzki** [reply]
Check this https://www.cloudflare.com/developer-platform/products/email-routing/ its free

**Reply by mikolajzawadzki** [reply]
And if you also want to send emails and not just receive them, maybe look for mxroute. I have been using mxroute for several years. They have a lifetime plan and there should be coupons for a 50% discount. You can connect multiple domains, probably without limits

**Reply by bparlan** [reply]
thanks for direction, i will check cloudflare routing for sure. yet regarding your second reply, i wanna clarify that my intention is not sending mail, being able to receive mails to my domain address.

**Reply by rootacc3ss** [reply]
I’m so sorry that I just saw this but no email thru Cloudflared or Cloudflare Tunnels. Hope you've found the solution since then.

**Comment by erdemozveren** [comment]
I was able to set up my stalwart mail server in coolify with docker compose with this guide https://www.aldertvaandering.com/blog/setting-up-stalwart-on-coolify/ and after i add necessary dns records to my domain i got 10/10 in mail-tester.com. Since docker compose file provided in guide i think it can be converted to one-click but i am new to coolify and dont know 'how to'

**Reply by graysham** [reply]
I followed this guide and Got errors about endpoints not being available from traefik end, Did you edit the guide at all?

As far as i can tell there seems to be some stuff missing, proxy protocol is enabled at stalwart end but there are no entrypoints or proxy protcol enabled at traefik end, or do i misunderstand something?

**Reply by erdemozveren** [reply]
Sorry i forgot to mention a typo in post,i did not get any errors after fixing this typo and rebuild
in section 'Configuring Stalwart to use the certs'

> Then add the end of the file add the following lines:


```
certificate.default.cert = "%{file:/data/certs/mail.YOUR_DOMAIN.com/cert.pem}%"
certificate.default.default = true
certificate.default.private-key = "%{file:/data/certs/mail.YOUR_DOMAIN.com/key.pem}>
```
this part have typo and should be like this
```
certificate.default.cert = "%{file:/data/certs/mail.YOUR_DOMAIN.com/cert.pem}%"
certificate.default.default = true
certificate.default.private-key = "%{file:/data/certs/mail.YOUR_DOMAIN.com/key.pem}%" <-- this is the change
```
and of course dont forget to update mail.YOUR_DOMAIN.com on all occurrences,
if your error is not because of this, idk what can be a cause.

**Reply by djsisson** [reply]
the guide has the container mapping the ports to the host, so this completely bypasses the proxy, so there is no need to add the entrypoints

the other way is you need to add all the entrypoints to the static proxy config, and then map the proxy to the host, and remove all the port mappings from the container

**Reply by graysham** [reply]
@djsisson that’s what I thought, what would be the benefits/draws to this method versus setting up with traefik doing reverse proxy and making use of proxy protocol.

**Reply by djsisson** [reply]
you can add some middleware like fail2ban within traefik etc as well as things like loadbalancing

**Reply by djsisson** [reply]
at the moment the suddomains arent really being used, you can access using anything that points to that serverip

going through the proxy you can then use the subdomains correctly

**Reply by graysham** [reply]
Reason I ask is I’ve been looking between this setup and stalwarts documentation on using traefik and I can’t seem to get the example in these docs to work properly. Any chance you could have a Quick Look? 

https://stalw.art/docs/server/reverse-proxy/traefik/

**Reply by djsisson** [reply]
did you add the port mappings in traefik, as well as the extra entrypoints for those ports?

this has to be done in a static config, so for coolify thats in the command section

**Reply by graysham** [reply]
Yes I added them in the static config, do you need to route the endpoints to load balancer as I noticed the stalwart compose relies on load balancer (sorry if this is a dumb question I’m just getting my head round the way traefik/coolify work). Also for proxy protocol is trusted IPs required or can you omit those lines (obviously best in production but whilst I’m diagnosing faults).

**Reply by djsisson** [reply]
as long as you have

```
--entrypoints.smtp.address=:25
--entrypoints.smtps.address=:465
--entrypoints.imaps.address=:993
```

im guessing you would need to add the coolify proxy subnet as trusted

also if you use a new network in your compose, traefik isn't added to it by default so i'ts inaccessible, so either you need to add traefik to your named network, or remove altogether and let coolify/docker generate one, coolify will then auto add traefik to it

**Reply by graysham** [reply]
I was ignoring the new network part and just using the coolify network or does it require a Seperate network to route between the two?

**Reply by IAmBlackat** [reply]
Unsupported server version
This webadmin release requires Stalwart Mail Server version 0.13.0 or later. Your server is running version 0.11.8.
any solution for this @erdemozveren

**Reply by erdemozveren** [reply]
I did not encounter this error, my versions if it helps is coolify [v4.0.0-beta.420.6], stalwartlabs/mail-server:v0.11 @IAmBlackat

**Reply by IAmBlackat** [reply]
> I did not encounter this error, my versions if it helps is coolify [v4.0.0-beta.420.6], stalwartlabs/mail-server:v0.11 @IAmBlackat

I was able to make it work, thank you so much. (he moved to new docker, old has been archived)
I still have some issue, if it is okay I would like to know how do you make it so it saves the data even if I restart the project?
Whenever I restart the project
<img width="621" height="162" alt="image" src="https://github.com/user-attachments/assets/dcba881a-63a7-450b-b679-7fdcd44b3339" />
from this top buttons, it clears everything and provide new admin password in the logs. (also removed all of my setup).
is there any solution for this?

**Reply by codetective** [reply]
> Unsupported server version This webadmin release requires Stalwart Mail Server version 0.13.0 or later. Your server is running version 0.11.8. any solution for this @erdemozveren

solved here:
 change the docker image to stalwartlabs/stalwart:latest
https://github.com/coollabsio/coolify/discussions/1142?utm_source=chatgpt.com#discussioncomment-14115510

**Reply by erdemozveren** [reply]
Sorry for late response @IAmBlackat , docker compose file has volumes so it doesn't lose data on restart, can you check your docker compose,there must be a similar `volumes` section similar to this (paths may be diffrent), so far nothing is losed as far as i know.Other than this i have no idea.
```yaml
    volumes:
      - '/var/lib/stalwart-mail:/opt/stalwart-mail'
      - /etc/localtime:/etc/localtime:ro
      - /data/coolify/certs:/data/certs:ro
 ```

**Reply by IAmBlackat** [reply]
Thank you so much, yes I found the path. really appreciate your help in this! 😊 @erdemozveren  @codetective

**Reply by rootacc3ss** [reply]
Just as a note, there seem to still be data persistence problems, and I had to use a different docker image than the one outlined in the article as it wasn't available. DYOR

EDIT: Following the changes, fixing the typo as @erdemozveren pointed out, and changing the version **FROM LATEST FULL STALWART PACKAGE (NOT MAIL-SERVER), WHICH UNFORTUNATELY SEEMS BROKEN WITH THE NEEDED DOCKER COMPOSE** (I am using v0.11 as mentioned), I have FINALLY reached stability and persistence.

Cheers, Erdem. Glory to the Turkish!

Also I encountered a bug trying to change admin pw to what I desired in the GUI --  It's probably best to do this anyhow, but I used "openssl passwd -6" to create a hashed PW and appended it to config.toml (for version 0.11 hash must be in quotes like username btw) and it is working!

**Reply by rootacc3ss** [reply]
@IAmBlackat now I am randomly receiving the error you got. Could you provide me with your fix? I will edit this message if I resolve "Unsupported server version" error...

**Reply by erdemozveren** [reply]
> @IAmBlackat now I am randomly receiving the error you got. Could you provide me with your fix? I will edit this message if I resolve "Unsupported server version" error...

I found solution
plese notice i still use `stalwartlabs/mail-server:v0.11.8-alpine` and this is works for me
if you use diffrent version of stalwart check release in here to find yours: https://github.com/stalwartlabs/webadmin/releases
i just look dates at docker container and webadmin relases and choose close one :D
open your config.toml and add or edit this configs,
```
# This is important, latest versions not compatible with my stalwart version
webadmin.resource = "https://github.com/stalwartlabs/webadmin/releases/download/v0.1.25/webadmin.zip"
webadmin.auto-update = true
```
then restart your container
it should revert to webadmin v0.1.25 , if everything is correct then go back to config.toml and update `webadmin.auto-update = false` and restart container again so it doesn't try to download it on every restart,(you may keep it to true if you want)
this sould not broke your data because you could just change/remove `webadmin.resource` to latest and get the latest version again but I do not accept responsibility.

btw: i switch to dokploy while ago because of my needs,so i can't answer for coolify further.
doc ref: https://stalw.art/docs/management/webadmin/update/

**Reply by IAmBlackat** [reply]
thank you very much! it worked and I am also using millionmail which is very easy to setup as well it is very good to manage the emails.

**Reply by mast3rp1ece** [reply]
I followed the steps here: https://www.aldertvaandering.com/blog/setting-up-stalwart-on-coolify/
 and used this image: stalwartlabs/mail-server:v0.11.8-alpine. However, I can't send mail; I'm getting this error: TLS error from 'alt4.gmail-smtp-in.l.google.com': STARTTLS not advertised by host. Has anyone faced this issue?

**Reply by djsisson** [reply]
google requires xoauth for smtp authentication

**Reply by mast3rp1ece** [reply]
But it's not only with gmail, but with every possible email, even when i trying to send mail to mail-tester.com, i'm getting same error

**Reply by djsisson** [reply]
where are you hosting?

**Reply by mast3rp1ece** [reply]
VPS Quasar Namecheap

**Reply by djsisson** [reply]
check they dont block or interfere with smtp

**Reply by mast3rp1ece** [reply]
it's not i think:
[root@mail ~]# telnet alt4.gmail-smtp-in.l.google.com 25
Trying 64.233.177.27...
Connected to alt4.gmail-smtp-in.l.google.com.
Escape character is '^]'.
220 mx.google.com ESMTP 00721157ae682-785aedf6c5esi30530287b3.10 - gsmtp

**Reply by mast3rp1ece** [reply]
✅ Solved for me.
Set it to disabled in Stalwart UI — that fixed the error.
Settings > SMTP > Outbound > TLS: STARTTLS → disabled.
<img width="860" height="182" alt="Screenshot 2025-10-24 134224" src="https://github.com/user-attachments/assets/190192f8-2418-4cbe-b45b-f39b11b302e0" />

**Reply by OSZII** [reply]
for those using the newer stalwart version, im still not done with installation and testing, but to persist the data between restarts you need to change the volume path:

old line:
'/var/lib/stalwart-mail:/opt/stalwart-mail'

New line
'/var/lib/stalwart-mail:/opt/stalwart'

this path got changed you can see it when you restart the server 
<img width="755" height="137" alt="image" src="https://github.com/user-attachments/assets/45ebebcd-4c2c-4280-b69e-76e16819e69b" />

the image is from the docs.

ill provide an update when im done!

**Comment by fsdevriver** [comment]
I installed it on my Coolify, but unfortunately, after one or two days, it shows a bad gateway. There is no log in the log tab.

**Reply by rootacc3ss** [reply]
This sounds like an issue I had with Coolify itself and what I believe was a messy docker configuration. Does it, by chance, continue to persistently show the bad gateway error even when you delete and redeploy the exact same thing? Same everything?

If so, check your Docker configuration or networking / Coolify Proxy for issues. 

Also, you shouldn't or can't be using CloudFlare tunnels for this. No email allowed on CloudFlare tunnels / cloudflared unless paying for enterprise, I fear.

**Comment by Malik-Ahsan-wq** [comment]
would be great as one click possible

**Comment by manstfu** [comment]
Weee neeed thisss!!!

**Comment by Cinzya** [comment]
Has been added as a Service now.
